File: src/main/java/com/rambots4571/rampage/hardware/DoubleSolenoid.java
package com.rambots4571.rampage.hardware;

/**
 * Wrapper class for {@link edu.wpi.first.wpilibj.DoubleSolenoid} allows
 * to easily toggle between values
 * {@link edu.wpi.first.wpilibj.DoubleSolenoid.Value}
 */
public class DoubleSolenoid extends edu.wpi.first.wpilibj.DoubleSolenoid {
    public DoubleSolenoid(int forwardChannel, int reverseChannel) {
        super(forwardChannel, reverseChannel);
    }

    public final void toggle() {
        Value currentValue = this.get() == Value.kForward ?
                             Value.kReverse : Value.kForward;
        this.set(currentValue);
    }
}

File: src/main/kotlin/com/rambots4571/rampage/hardware/DoubleSolenoid.kt
package com.rambots4571.rampage.hardware

import edu.wpi.first.wpilibj.DoubleSolenoid

class DoubleSolenoid(forwardChannel: Int, reverseChannel: Int) :
        DoubleSolenoid(forwardChannel, reverseChannel) {

    fun toggle() {
        val currentValue =
            if (get() == Value.kForward) Value.kReverse else Value.kForward
        set(currentValue)
    }
}